### PR TITLE
[Feature] Add methods for async task

### PR DESCRIPTION
**Pull request by SferaDev**
Pending:

- get/getAll split on models too

### CODE REVIEW

**Comment by adrianq** [comment]
@tokland I have tested the PR that is using this feature in MDSync and seems to be working fine. I will wait for your review before merging

File: src/api/system.ts
Comment (tokland) [inline src/api/system.ts:55]: [minor] according to the type, we don't need `|| []`, it may be undefined or null? we should add that possibility also to the type. In any case the check `!result! in the next line is always true, right?

**Review by tokland — APPROVED** [review]
[code review, no feature testing] 

Cool PR, a lot of new features! Let's try to have `api.models.getAll` on the next iteration.

About the async code, it's very nice to have typed `TaskCategory` and mapped types in `WaitForResponse`, it must be a joy to use as a caller, cool!

File: src/api/system.ts
Comment (SferaDev) [reply src/api/system.ts]:
```ts
            const checkTask = async () => {
            const result = await this.d2Api
                .get<{ message: string; completed?: boolean }[]>(`/system/tasks/${jobType}/${id}`)
                .getData();

            if (!Array.isArray(result) || !result[0]) return false;
            return result[0].completed;
        };
```

**Review by tokland — APPROVED** [review]

**Pull request merged**
